cli/src/main/kotlin/commands/UploadResultToSw360Command.kt

```diff
@@ -0,0 +1,184 @@
+            "command expects that a Sw360Storage for the scanner is configured."
+) {
+    private val ortFile by option(
+        "--ort-file", "-i",
+        help = "The ORT result file to read as input."
+    ).convert { it.expandTilde() }
+        .file(mustExist = true, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = true)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+        .inputGroup()
+
+    private val sw360JsonMapper = jsonMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+    private val globalOptionsForSubcommands by requireObject<GlobalOptions>()
+
+    override fun run() {
+        val (ortResult, duration) = measureTimedValue { ortFile.readValue<OrtResult>() }
+
+        log.perf {
+            "Read ORT result from '${ortFile.name}' (${ortFile.formatSizeInMib}) in ${duration.inMilliseconds}ms."
+        }
+
+        val sw360Config = globalOptionsForSubcommands.config.scanner?.storages?.values
+            ?.filterIsInstance<Sw360StorageConfiguration>()?.singleOrNull()
+
+        requireNotNull(sw360Config) {
+            "No SW360 storage is configured for the scanner."
+        }
+
+        val sw360Connection = createSw360Connection(sw360Config, sw360JsonMapper)
+        val sw360ReleaseClient = sw360Connection.releaseAdapter
+        val sw360ProjectClient = sw360Connection.projectAdapter
+
+        getProjectWithPackages(ortResult).forEach { (project, pkgList) ->
+            val linkedReleases = pkgList.mapNotNull { pkg ->
+                val name = listOfNotNull(pkg.id.namespace, pkg.id.name).joinToString("/")
+                sw360ReleaseClient.getSparseReleaseByNameAndVersion(name, pkg.id.version)
+                    .flatMap { sw360ReleaseClient.enrichSparseRelease(it) }
+                    .orElse(createSw360Release(pkg, sw360ReleaseClient))
+            }
+
+            val sw360Project = sw360ProjectClient.getProjectByNameAndVersion(project.id.name, project.id.version)
+                .orElse(createSw360Project(project, sw360ProjectClient))
+
+            sw360Project?.let {
+                sw360ProjectClient.addSW360ReleasesToSW360Project(it.id, linkedReleases)
+            }
+        }
+    }
+
+    private fun createSw360Project(project: Project, client: SW360ProjectClientAdapter): SW360Project? {
+        val sw360Project = SW360Project()
+            .setName(project.id.name)
+            .setVersion(project.id.version)
+            .setDescription("A ${project.id.type} project with the PURL ${project.id.toPurl()}.")
+            .setVisibility(SW360Visibility.BUISNESSUNIT_AND_MODERATORS)
+
+        return try {
+            val createdProject = client.createProject(sw360Project)
+            createdProject?.let {
+                log.debug { "Project ${createdProject.name}-${createdProject.version} created in SW360." }
+            }
+
+            createdProject
```

Comment:
I believe this could be replaced with

            client.createProject(sw360Project)?.also {
                log.debug { "Project '${it.name}-${it.version}' created in SW360." }
            }

Note that I also single-quoted the project for readability.